src/HelixServer.js

```diff
@@ -86,24 +86,23 @@ function executeTemplate(ctx) {
   });
 
   Module._nodeModulePaths = nodeModulePathsFn;
-  const actionArgs = {
+
+  return Promise.resolve(mod.main({
     __ow_headers: owHeaders,
     __ow_method: ctx.method.toLowerCase(),
     __ow_logger: ctx.logger,
     owner: ctx.config.contentRepo.owner,
     repo: ctx.config.contentRepo.repo,
     ref: ctx.config.contentRepo.ref || 'master',
     path: `${ctx.resourcePath}.md`,
+    selector: ctx._selector,
+    extension: ctx._extension,
+    url: ctx._url,
```

Comment:
please note that url is not sent by fastly in VCL. so I wouldn't include it here

Reply:
Ok, but I guess we should also remove it from [request.schema.json](https://github.com/adobe/helix-pipeline/blob/master/docs/request.schema.json) then...

@trieloff

Reply:
see: https://github.com/adobe/helix-cli/blob/master/layouts/fastly/helix.vcl#L317

Reply:
I would generate the URL request property in the pipeline (in openwhisk.js)

Reply:
Or add it in `helix.vcl` :)

Should we add `strain` here (and in the pipeline) instead? Would it be useful for developers?

Reply:
> I would generate the URL request property in the pipeline (in openwhisk.js)

ok

Reply:
> Should we add strain here (and in the pipeline) instead? Would it be useful for developers?

we don't use strains yet in simulator. you can add `strain=default` for completness.

Reply:
> Or add it in helix.vcl :)

the URL is easily reconstructable with `path` and `param`. no need to add it twice.

Reply:
yes, and selector and extension

Reply:
> we don't use strains yet in simulator. you can add `strain=default` for completness.

`strain` isn't mentioned in `request.schema.json`, so not sure if we really need it